File: Baekjoon/1912.py
import sys 
input = sys.stdin.readline 

n = int(input())
sequence = list(map(int, input().split()))
memo = [0]*(n)
memo[0] = sequence[0]

for i in range(1,n):
    if memo[i-1] + sequence[i] >= sequence[i]:
        memo[i] = memo[i-1] + sequence[i] 
    else:
        memo[i] =  sequence[i] 


print(max(memo))

File: JavaScript/Baekjoon/1912.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = fs.readFileSync(filePath).toString().split("\n");

const n = parseInt(input[0]);

const numbers = input[1].split(" ").map((item) => parseInt(item));

let dp = Array.from({ length: n }, () => 0);

dp[0] = numbers[0];

for (let i = 1; i < n; i++) {
  dp[i] = Math.max(dp[i - 1] + numbers[i], numbers[i]);
}

console.log(Math.max(...dp));
